File: Top 100 Codes/02-Working with Numbers/03-Greatest Common Divisor/GreatestCommonDivisor.py
# Function to calculate the Greatest Common Divisor (GCD) of two integers
def gcd(a, b):
    # Continue the loop until one of the numbers becomes zero
    while b != 0:
        remainder = a % b  # Find the remainder when 'a' is divided by 'b'
        a = b              # Assign the value of 'b' to 'a'
        b = remainder      # Assign the remainder to 'b'
    return a  # 'a' contains the GCD when 'b' becomes zero

# Input two numbers from the user
num1 = int(input("Enter first integer: "))
num2 = int(input("Enter second integer: "))

# Output the GCD of the two numbers
print(f"The GCD of {num1} and {num2} is {gcd(num1, num2)}")

File: Top 100 Codes/02-Working with Numbers/03-Greatest Common Divisor/GreatestCommonDivisor.cpp
#include <iostream>
using namespace std;

// Function to calculate the Greatest Common Divisor (GCD) of two integers
int gcd(int a, int b) {
    // Continue the loop until one of the numbers becomes zero
    while (b != 0) {
        int remainder = a % b; // Find the remainder when 'a' is divided by 'b'
        a = b;                // Assign the value of 'b' to 'a'
        b = remainder;        // Assign the remainder to 'b'
    }
    return a; // 'a' contains the GCD when 'b' becomes zero
}

int main() {
    int num1, num2;

    // Input two numbers from the user
    cout << "Enter two integers: ";
    cin >> num1 >> num2;

    // Output the GCD of the two numbers
    cout << "The GCD of " << num1 << " and " << num2 << " is " << gcd(num1, num2) << endl;

    return 0;
}
